.github/workflows/macos_tests.yml

```diff
@@ -0,0 +1,157 @@
+name: macOS tests
```

Comment:
Why do we provide a separate workflow for this instead of inside the existing testing workflow? We can disable it by default there but in the end all of our projects want to test on macOS and it would be easier to roll it out wide without requiring a new workflow. Maybe I'm missing something though.

Reply:
I tried to give a nod to this in the description but perhaps more detail is needed:
> This functionality is implemented as a standalone workflow to preserve flexibility as requirements and capacity change over time.

The pattern of use (on PRs?, run nightly?), requirements, and node capacity are likely to be in flux over the short term as more repositories are onboarded and we understand load - I wanted to reserve as much flexibility as possible. I agree in an ideal world they'd be one matrix.

Reply:
The actual task(s) we're performing here is also very different, we want to run a bunch of checks on one node to avoid the recycle time, not just `swift test`.

Reply:
Makes sense. Thanks for expanding